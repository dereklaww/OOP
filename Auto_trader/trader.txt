trader1: $375.00 {}
trader2: $25.00 {ABC: 10.00}
trader3: $50.00 {ABC: 5.00}
trader4: $100.00 {}
trader5: $-50.00 {ABC: 20.00}
