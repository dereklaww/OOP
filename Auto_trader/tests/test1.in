add trader -100
add trader 100
add trader 0
balance trader
balance trader_2
inventory trader_2
inventory trader 
import trader Nike 0
import trader Nike 100
import trader_2 Nike 100
amount trader Nike
amount trader Adidas
amount trader_2 Nike
export trader Nike 50
export trader Adidas 100
export trader_2 Nike 100
exit