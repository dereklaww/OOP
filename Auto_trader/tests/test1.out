Initial balance cannot be negative.
Success.
Trader with given ID already exists.
$100.00
No such trader in the market.
No such trader in the market.
Trader has an empty inventory.
Could not import product into market.
Trader now has 100.00 units of Nike.
No such trader in the market.
100.00
Product not in inventory.
No such trader in the market.
Trader now has 50.00 units of Nike.
Could not export product out of market.
No such trader in the market.
Have a nice day.